.github/workflows/cuda.yml

```diff
@@ -0,0 +1,54 @@
+name: CUDA Build Checks
```

Comment:
I wonder if we should just have a single build with all HMEM configuration toggled on. We don't gain much by individually compiling for rocm/cuda/ze.

Reply:
Installing cuda already takes about 7 minutes just to install. I know the oneapi lib also takes a while to install. Maybe one of each in parallel?

Reply:
@rajachan - I agree.  Do you know what the time constraints are?

Reply:
We can run up to 20 jobs in parallel, with each workflow taking up to 72 hours before it gets terminated, so we have plenty of room with the service. The only constraint is what we enforce on ourselves for CI completion time for each PR. These actions won't be the long pole, so it might be okay for now to install the different SDKs serially. I'm surprised even a binary install of CUDA from the debian repo is taking that long, although I've never paid attention to the time.

From what I can tell, different "jobs" or "workflows" can be run in parallel, but different "steps" can not. So we can't really have a parallel install of each SDK within a job and then a fence to run the configure/make step. If the runtime ends up becoming a problem in the future, we can just bake all dependencies, including rdma-core, into a container and use that for `runs-on`.